core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -440,6 +444,9 @@ private[spark] class MesosClusterScheduler(
         .mkString(",")
       options ++= Seq("--py-files", formattedFiles)
     }
+    desc.schedulerProperties
+      .filter { case (key, _) => !replicatedOptionsBlacklist.contains(key) }
+      .foreach { case (key, value) => options ++= Seq("--conf", Seq(key, "=\"", value, "\"").mkString("")) }
```

Comment:
Yes, the quotes work for me.

In order to pass multiple system properties via eg `spark.driver.extraJavaOptions`, I passed the arg in quotes on the shell to spark-submit, but the value itself doesn't have quotes around it and ends up broken on the commandline for the driver, unless I add escaped quotes inside the string.

So this fails when executing the driver, but spark-submit accepts the command:
```
bin/spark-submit [...] spark.driver.extraJavaOptions="-Dcom.sun.jersey.server.impl.cdi.lookupExtensionInBeanManager=true -Dorg.jboss.weld.bootstrap.concurrentDeployment=false" [...]
```
This doesn't look very intuitive to me, but it does work without wrapping values in quotes:
```
bin/spark-submit [...] spark.driver.extraJavaOptions="\"-Dcom.sun.jersey.server.impl.cdi.lookupExtensionInBeanManager=true -Dorg.jboss.weld.bootstrap.concurrentDeployment=false\"" [...]
```

String interpolation doesn't support double quotes, but I've since learned about triple-double-quoted string interpolation, that why I initially went with `Seq.mkString`.